### ISSUE TITLE
Four-button switch TS0601 not supported.

### ISSUE THREAD

**Issue by mdanilohome** [Koenkk/zigbee-herdsman-converters#4644]
Friend could help me with this device more? I purchased this four-button switch provided by Shawader, model SMKG-4KNL-US/TZB-W and when trying to pair it with zigbee 2mqtt it appears that it is not supported. Do you need an external converter? If so, could you help me with the creation? Thank you very much in advance.
![Print Screen Zigbee 2mqtt](https://user-images.githubusercontent.com/86743538/189434104-30095104-760a-451f-b9fc-c680b4882547.jpg)

**Comment by mdanilohome** [comment]
I tried creating and installing an external converter using one of the examples and zigbee 2mqtt stopped working. The following errors appeared:

[17:29:33] INFO: Handing over control to Zigbee2MQTT Core ...
Error: Cannot find module 'TZE200_6wi2mope'
Require stack:
- /app/dist/util/utils.js
- /app/dist/util/settings.js
- /app/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.getExternalConvertersDefinitions (/app/lib/util/utils.ts:167:25)
    at getExternalConvertersDefinitions.next (<anonymous>)
    at new ExternalConverters (/app/lib/extension/externalConverters.ts:12:20)
    at new Controller (/app/lib/controller.ts:84:58)
    at start (/app/index.js:108:18)
[17:29:43] INFO: Handing over control to Zigbee2MQTT Core ...
Error: Cannot find module 'TZE200_6wi2mope'
Require stack:
- /app/dist/util/utils.js
- /app/dist/util/settings.js
- /app/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.getExternalConvertersDefinitions (/app/lib/util/utils.ts:167:25)
    at getExternalConvertersDefinitions.next (<anonymous>)
    at new ExternalConverters (/app/lib/extension/externalConverters.ts:12:20)
    at new Controller (/app/lib/controller.ts:84:58)
    at start (/app/index.js:108:18)

**Comment by mdanilohome** [comment]
This was the .js file I added in zigbee 2mqtt:

const exposes = require('../lib/exposes');
const fz = {...require('../converters/fromZigbee'), legacy: require('../lib/legacy').fromZigbee};
const tz = require('../converters/toZigbee');
const ota = require('../lib/ota');
const tuya = require('../lib/tuya');
const reporting = require('../lib/reporting');
const extend = require('../lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const zosung = require('../lib/zosung');
const utils = require('../lib/utils');
const fzZosung = zosung.fzZosung;
const tzZosung = zosung.tzZosung;
const ez = zosung.presetsZosung;

const fzLocal = {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_6wi2mope'}],
        model: 'SMKG-4KNL-US/TZB-W',
        vendor: 'Shawader',
        description: 'Wall touch light switch (4 gang)',
        exposes: [e.switch().withEndpoint('l1').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l2').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l3').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l4').setAccess('state', ea.STATE_SET),
            exposes.enum('indicate_light', ea.STATE_SET, Object.values(tuya.moesSwitch.indicateLight))
                .withDescription('Indicator light status'),
            exposes.enum('power_on_behavior', ea.STATE_SET, Object.values(tuya.moesSwitch.powerOnBehavior))
                .withDescription('Controls the behavior when the device is powered on')],
        fromZigbee: [fz.ignore_basic_report, fz.tuya_switch, fz.moes_switch],
        toZigbee: [tz.tuya_switch_state, tz.moes_switch],
        onEvent: tuya.onEventSetLocalTime,
        meta: {multiEndpoint: true},
        endpoint: (device) => {
            // Endpoint selection is made in tuya_switch_state
            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1};
        },
        configure: async (device, coordinatorEndpoint, logger) => {
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(2)) await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(3)) await reporting.bind(device.getEndpoint(3), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(4)) await reporting.bind(device.getEndpoint(4), coordinatorEndpoint, ['genOnOff']);
            // Reports itself as battery which is not correct: https://github.com/Koenkk/zigbee2mqtt/issues/6190
            device.powerSource = 'Mains (single phase)';
            device.save();
        },
    }

module.exports = definition;

**Comment by mdanilohome** [comment]
I made a correction in the name of the file that was missing the .js in the configuration.yaml of zigbee 2mqtt. The error log now appears as below:

[18:38:57] INFO: Handing over control to Zigbee2MQTT Core ...
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '../lib/exposes'
Require stack:
- /app/dist/util/utils.js
- /app/dist/util/settings.js
- /app/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at /app/dist/util/externally-loaded.js:1:17
    at Script.runInContext (node:vm:139:12)
    at Script.runInNewContext (node:vm:144:17)
    at Object.runInNewContext (node:vm:298:38)
    at loadModuleFromText (/app/lib/util/utils.ts:148:8)
    at loadModuleFromFile (/app/lib/util/utils.ts:155:12)

**Comment by ciedema** [comment]
Start here https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_1-pairing-the-device-with-zigbee2mqtt

Look the difference between your code, and the examples.

**Comment by mdanilohome** [comment]
Hi @ciedema , thank you.  I built this converter using as reference the codes contained in the Moes link https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/moes.js that are equivalent to my device, a four-button switch and when paired, my device remains as "not supported".   
Are there any corrections or modifications that still need to be made? Would you help me?
I made the changes to include my device specs and created the converter below, but as I said, the device is still not supported:

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const zosung = require('zigbee-herdsman-converters/lib/zosung');
const utils = require('zigbee-herdsman-converters/lib/utils');
const fzZosung = zosung.fzZosung;
const tzZosung = zosung.tzZosung;
const ez = zosung.presetsZosung;

const fzLocal = {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_6wi2mope'}],
        model: 'SMKG-4KNL-US/TZB-W',
        vendor: 'Shawader',
        description: 'Wall touch light switch (4 gang)',
        exposes: [e.switch().withEndpoint('l1').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l2').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l3').setAccess('state', ea.STATE_SET),
            e.switch().withEndpoint('l4').setAccess('state', ea.STATE_SET),
            exposes.enum('indicate_light', ea.STATE_SET, Object.values(tuya.moesSwitch.indicateLight))
                .withDescription('Indicator light status'),
            exposes.enum('power_on_behavior', ea.STATE_SET, Object.values(tuya.moesSwitch.powerOnBehavior))
                .withDescription('Controls the behavior when the device is powered on')],
        fromZigbee: [fz.ignore_basic_report, fz.tuya_switch, fz.moes_switch],
        toZigbee: [tz.tuya_switch_state, tz.moes_switch],
        onEvent: tuya.onEventSetLocalTime,
        meta: {multiEndpoint: true},
        endpoint: (device) => {
            // Endpoint selection is made in tuya_switch_state
            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1};
        },
        configure: async (device, coordinatorEndpoint, logger) => {
            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(2)) await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(3)) await reporting.bind(device.getEndpoint(3), coordinatorEndpoint, ['genOnOff']);
            if (device.getEndpoint(4)) await reporting.bind(device.getEndpoint(4), coordinatorEndpoint, ['genOnOff']);
            // Reports itself as battery which is not correct: https://github.com/Koenkk/zigbee2mqtt/issues/6190
            device.powerSource = 'Mains (single phase)';
            device.save();
        },
    }

module.exports = definition;

**Comment by mdanilohome** [comment]
Follow the LOG

![Log device](https://user-images.githubusercontent.com/86743538/189493085-d10d4af9-347b-4da8-93d5-8a69a4c89ae1.jpg)

**Comment by ciedema** [comment]
Have you added the convertor into configuration.yaml and is it loading correctly?

**Comment by mdanilohome** [comment]
Hi, thanks for the reply. See the image, I believe that everything is correct, both the file with the codes in the correct place, and the line of code from the external converter in the configuration.yaml .
![External Converter em configuration yaml](https://user-images.githubusercontent.com/86743538/189529525-029edaa1-b83a-4337-bc87-d9b550b62aec.jpg)

**Comment by mdanilohome** [comment]
Log during pairing:
I identified that the device is paired, but I could not identify in this log when the external converter is loaded.
![Log during pairing](https://user-images.githubusercontent.com/86743538/189530396-b1ff3a4a-9e74-492d-ad06-2895f2b6f4f8.jpg)

**Comment by mdanilohome** [comment]
I added the external convert line of code through the screen below, restarted zigbee 2mqtt and it crashed on reboot. Previously I was typing the line of code directly into configuration.yaml and although it was not crashing zigbee 2mqtt, I noticed in the previous log that in this condition the external converter was not loading. In this other way, it started to load but presented another error in the log and zigbee 2mqtt crashed.
![Line of code inclusion log through z2m settings](https://user-images.githubusercontent.com/86743538/189531529-9843de6a-c9b6-432b-ace3-f645263d5483.jpg)
